wiringPi/wiringPi.c

```diff
@@ -1559,11 +1579,8 @@ void pwmSetRangeChannel (unsigned int range, unsigned int channel) {
         return;
       }
 
-      pwm[RP1_PWM0_CHAN0_RANGE] = range;
-      pwm[RP1_PWM0_CHAN1_RANGE] = range;
-      pwm[RP1_PWM0_CHAN2_RANGE] = range;
-      pwm[RP1_PWM0_CHAN3_RANGE] = range;
-      readback = pwm[RP1_PWM0_CHAN0_RANGE];
+      ((pwm_RP1)pwm)->CHAN[channel].RANGE = range;
+      readback = ((pwm_RP1)pwm)->CHAN[channel].RANGE;
```

Comment:
The main downside here is that I have to type cast `*pwm` to the struct type in order to use it. Shortest fix is to use a `#define` to mask the ugly type cast, but the cleanest and most future-proof way to do it might be to change the type of `*pwm` to a union of structs for the RP1 and BCM283X chips.

Obviously that would make this a much larger project since I would need to change all places where `*pwm` is used. I have the time to handle that, but I wanted to defer to you before steamrolling over other parts of currently-working code.

Thoughts?